app/family-history/family-history.service.js

```diff
@@ -0,0 +1,268 @@
+          when eligible_for_testing = 1065 then  @eligible_for_tracing:=2
+          else @eligible_for_tracing:=0 
+        end as eligible_for_tracing,
+        case 
+          when facility_enrolled is not null then facility_enrolled  
+        end as fm_facility_enrolled,
+        date_format(preferred_testing_date,"%d-%m-%Y") as preferred_testing_date
+      from etl.flat_family_testing where patient_uuid = '${params.patientUuid}'`;
+      /*
+      1.eligible_for_tracing = 1, traced and tested
+      2.eligible_for_tracing = 2, eligible for testing
+      3.eligible_for_tracing = 0, not eligible for testing 
+      */
+      queryParts = {
+        sql: sql
+      };
+      return db.queryServer(queryParts, function (result) {
+        result.sql = sql;
+        resolve(result);
+      });
+    });
+  };
+
+  updateRegisteredContact = (params) => {
+    return new Promise((resolve, reject) => {
+      let queryParts = {};
+      let sql =
+        'update etl.flat_family_testing set is_registered = 1, fm_uuid = "' +
+        params.uuid +
+        '" where obs_group_id = ' +
+        params.obs_group_id +
+        '';
+      queryParts = {
+        sql: sql
+      };
+      return db.queryServer(queryParts, function (result) {
+        result.sql = sql;
+        resolve(result);
+      });
+    });
+  };
+
+  saveContactTracing = (params) => {
+    return new Promise((resolve, reject) => {
+      let queryParts = {};
+      let sql = '';
+
+      if (params.query.trace_id != null) {
+        sql =
+          'update etl.contact_tracing set contact_id = ' +
+          params.payload.contact_id +
+          ',contact_date="' +
+          params.payload.contact_date +
+          '",contact_type=' +
+          params.payload.contact_type +
+          ',contact_status=' +
+          params.payload.contact_status +
+          ',reason_not_contacted=' +
+          params.payload.reason_not_contacted +
+          ',remarks="' +
+          params.payload.remarks +
+          '" where id = ' +
+          params.query.trace_id +
+          '';
+      } else {
+        sql =
+          'insert into etl.contact_tracing (contact_id,contact_date,contact_type,contact_status,reason_not_contacted,remarks) values( ' +
+          params.payload.contact_id +
+          ',"' +
+          params.payload.contact_date +
+          '",' +
+          params.payload.contact_type +
+          ',' +
+          params.payload.contact_status +
+          ',' +
+          params.payload.reason_not_contacted +
+          ',"' +
+          params.payload.remarks +
+          '")';
+      }
+      queryParts = {
+        sql: sql
+      };
+      return db.queryServer(queryParts, function (result) {
+        result.sql = sql;
+        resolve(result);
+      });
+    });
+  };
+
+  getContactTracingHistory = (params) => {
+    return new Promise((resolve, reject) => {
+      let queryParts = {};
+      let sql = `select t2.*, id, contact_id, contact_date, remarks,
```

Comment:
No need for when @contact_type variable since its not being carried forward. 
when contact_type = 1555 then 'Phone tracing'  should be enough